app/gui/src/presenter.rs

```diff
@@ -125,11 +138,37 @@ impl Model {
                 } else {
                     warn!("Project creation failed: no ProjectManagingAPI available.");
                 }
-            })
+            }))
         } else if let Some(template) = template {
             error!("Invalid project template name: {template}");
         };
     }
+
+    /// Open a project by name or ID. If no project with the given name exists, it will be created.
+    #[profile(Task)]
+    fn open_or_create_project(&self, project: ProjectToOpen) {
+        let controller = self.controller.clone_ref();
+        crate::executor::global::spawn(with_progress_indicator(|| async move {
+            if let Ok(managing_api) = controller.manage_projects() {
+                if let Err(error) = managing_api.open_or_create_project(project).await {
+                    error!("Cannot open or create project: {error}.");
+                }
+            } else {
+                warn!("Project opening failed: no ProjectManagingAPI available.");
```

Comment:
What is "ProjectManagingAPI" - is this a name of a struct? I don't see it imported anywhere. Anyway, errors should provide user-facing messages and putting there names of local structs does not make much sense.

Reply:
> What is "ProjectManagingAPI" - is this a name of a struct? I don't see it imported anywhere. Anyway, errors should provide user-facing messages and putting there names of local structs does not make much sense.

`ProjectManagingAPI` is returned by `manage_projects` function.

And I disagree: this is not a message for the user, but a log. The logs are not for users, but for developers to quickly realize what went wrong.

Here, the better message would be something like `The IDE controller has no ProjectManagingAPI defined`.

Reply:
I think the old version was even better and still provide the necessary information to the developer: `Project Manager API not available, cannot open project.`. Why putting the name of the struct in the error message is better than just creating a sentence as the previous version? I'm talking about it to find a common ground here and make a decision about similar situations for the future.

Reply:
I just took that line from the similar functions above it. I'm happy to change it to whatever. I don't think it's important at all. I'll change it here and in the functions above to "Project Manager API not available, cannot open project."